streamlit_app.py

```
@@ -0,0 +1,228 @@
+    }
+
+# Streamlit UI
+st.title("🔨 STL File Splitter")
+st.write("Upload an STL file and split it into smaller parts for 3D printing")
+
+# File uploader
+uploaded_file = st.file_uploader("Choose an STL file", type=['stl'])
+
+if uploaded_file is not None:
+    # Save uploaded file temporarily
+    temp_input = tempfile.NamedTemporaryFile(delete=False, suffix='.stl')
```

Ruff (0.8.2)

Use a context manager for opening files

(SIM115)